sequential/seq2pat.py

```diff
@@ -423,27 +464,28 @@ def get_patterns(self, min_frequency: Num) -> List[list]:
         # Check num_rows
         check_true(self._num_rows >= 1, ValueError("Sequences should not be empty."))
 
-        # Check min_frequency conditions
-        if isinstance(min_frequency, float):
-            check_true(0.0 < min_frequency,
-                       ValueError("Frequency percentage should be greater than 0.0", min_frequency))
-            check_true(min_frequency <= 1.0, ValueError("Frequency percentage should be less than 1.0", min_frequency))
-            check_true(min_frequency * self._num_rows >= 1.0, ValueError("Frequency percentage should set the minimum "
-                                                                         "row count to be no less than 1.0."
-                                                                         "Thus the percentage should be no less than "
-                                                                         "1/(number of sequences)."))
-        elif isinstance(min_frequency, int):
-            check_true(0 < min_frequency, ValueError("Frequency should be greater than 0.0", min_frequency))
-            check_true(min_frequency <= self._num_rows, ValueError("Frequency cannot be more than number of sequences ",
-                                                                   min_frequency))
-        else:
-            raise TypeError("Frequency should be integer (as a row count) or float (as a row percentage)")
+        if not self.batch_size:
 
-        # Cython implementor object with input parameters set
-        self._cython_imp = self._get_cython_imp(min_frequency)
+            # Check min_frequency conditions
+            # Given a set of `num_rows` sequences, we need to validate min_frequency.
+            # In edge cases when min_frequency is not set properly, e.g. 0 or less than 1/num_rows, program will fail.
+            validate_min_frequency(self._num_rows, min_frequency)
 
-        # Frequent mining
-        patterns = self._cython_imp.mine()
+            # Call Cython backend in a child process, write results to queue
```

Comment:
are we gettign same results when nothing is used in here?

Reply:
Yes, we get the same results. The existing unit tests can get passed.